### ISSUE TITLE
Unable to create session

### ISSUE THREAD

**Issue by afzbkr** [Mastercard/gateway-ios-sdk#2]
Hi,
Edited the AppDelegate.swift to my test MID, region and merchant server. But Im getting Error: Unable to create session. What could be wrong? Thanks

**Comment by jmonroe** [comment]
Sorry you are having trouble.  So that I can help get to the bottom of the issue, would you mind answering the following questions...

Are you using the test merchant server we provide at https://github.com/Mastercard/gateway-test-merchant-server?

When you run the sample app in the simulator, what kind of messages are you seeing logged in the Xcode console?

Thanks

**Comment by afzbkr** [comment]
Hey.. yes im using the test merchant server provided. Im using MTF credentials and MTF as region. Would that be an issue?

There was nothing about error on the console. I debug and sessionId is Null hence the error.

**Comment by jmonroe** [comment]
I am sorry that these problems are persisting for you.

Your region and credentials sound right.

There were some changes back in January for both the mobile SDK and the test merchant server that enabled the test merchant server to respond to the app with the api version the session was created with.  Support was added to the mobile sdk also. 

From what you describe, It appears your test merchant server is not running the latest version.  Please verify that you are using the latest version.  Possibly, deploy it to a new Heroku instance just to be sure.

Thanks

**Comment by afzbkr** [comment]
you're right about using the latest ver of the test merchant server. Im now using the latest ver and is able to go to Enter Payment Details page. However im stuck again with Error: Unable to update session after filling up all the payment details and clicking continue.

**Comment by afzbkr** [comment]
error(Gateway Request Error - 400 bad request)
![screen shot 2018-02-20 at 2 32 38 pm](https://user-images.githubusercontent.com/25782354/36410316-3d6fd3fc-164b-11e8-8bf2-5e4691e39ea7.png)

**Comment by jmonroe** [comment]
Ok, I have pushed a hot fix that corrects a problem where the error message was not being propagated in the SDK correctly. If you get the latest version and run it again, you should get a more helpful error.

**Comment by afzbkr** [comment]
Hi. The error is due to Merchant 'TEST5004334' is not enabled to use the API

![screen shot 2018-02-21 at 11 37 25 am](https://user-images.githubusercontent.com/25782354/36462054-0d8f8478-16fc-11e8-8f49-ac49c6987483.png)

Thanks.

**Comment by afzbkr** [comment]
hey.. solved the above. it is due to merchant configuration. 

Im now faced with a different error which is "No value associated with key merchant (\"merchant\")."

![screen shot 2018-02-21 at 12 32 10 pm](https://user-images.githubusercontent.com/25782354/36466254-1acd9978-1714-11e8-963e-47843aaef3bf.png)

Thanks.

**Comment by jmonroe** [comment]
Sorry that you have run into another issue,

I believe that the API does not always return merchant so I have now made this optional.  Please download 0.8.2 and try again.

Thanks

**Comment by afzbkr** [comment]
Hi

Downloaded 0.8.2 and getting this error:

dataCorrupted(Swift.DecodingError.Context(codingPath: [MPGSDK_iOS_Sample.MerchantAPIResponse<MPGSDK_iOS_Sample.CompleteSessionResponse>.(CodingKeys in _6916ADD04A67DBC52AF6B4BC897DA2E2).gatewayResponse, MPGSDK_iOS_Sample.CompleteSessionResponse.(CodingKeys in _6916ADD04A67DBC52AF6B4BC897DA2E2).result], debugDescription: "Cannot initialize APIResult from invalid String value ERROR", underlyingError: nil))

![screen shot 2018-02-22 at 9 37 08 am](https://user-images.githubusercontent.com/25782354/36518225-9dad8e6e-17c0-11e8-88bc-ce8e5328db8f.png)

Thanks

**Comment by jmonroe** [comment]
Ok, I have gone a step further and updated the sample app to use the GatewayMap objects for payloads like we do in the main SDK.  This should go a long way in debugging the issue.  

If you download 0.8.3 and try again, I think you will have much better luck.

I am sorry that this has taken several updates to work thought.  The good news is that all the issues have been related to the sample app and not the SDK itself.  Thank you for your patience in working though these issues.

**Comment by afzbkr** [comment]
Hey that's alright. We are getting a step closer to the end at least.

Getting this error now:
["gatewayResponse": ["result": "ERROR", "error": ["validationType": "INVALID", "field": "transaction.frequency", "explanation": "Value \'SINGLE\' is invalid. Missing merchant privilege \'merchTxSrcSubType\' for merchant \'TEST5004334\'", "cause": "INVALID_REQUEST"]], "apiVersion": "44"]

I've double checked that my merchant configuration is pointing to frequency=Single.

When I simulated the payment process using Postman I was able to get payment approved using same payment info, amount and currency.

Thanks
